google-cloud-core/src/main/java/com/google/cloud/HttpTransportOptions.java

```diff
@@ -154,10 +156,27 @@ public void initialize(HttpRequest httpRequest) throws IOException {
         if (readTimeout >= 0) {
           httpRequest.setReadTimeout(readTimeout);
         }
+
+        HttpHeaders headers = httpRequest.getHeaders();
+        headers.set("x-goog-api-client", getXGoogApiClientHeader());
       }
     };
   }
 
+  String getXGoogApiClientHeader() {
+    return String.format(
+        "gl-java/%s %s/%s gax/%s",
```

Comment:
We may have a situation in the future where gax is used for certain http contexts, and in that case I think it will make sense. As of now, none of our http codepaths use gax.